src/aws-codeartifact/domain.ts

```diff
@@ -0,0 +1,213 @@
+}
+
+/**
+ * A new or imported clustered database.
+ */
+abstract class DomainBase extends Resource implements IDomain {
+  public abstract readonly domainArn: string;
+  public abstract readonly domainName: string;
+  public abstract readonly encryptionKeyArn?: string;
+  public abstract readonly domainOwner: string;
+}
+
+export interface DomainAttributes {
+  readonly domainArn: string;
+  readonly domainName: string;
+  readonly encryptionKeyArn?: string;
+  readonly domainOwner: string;
+}
+
+export interface DomainProps {
+  /**
+   * The name of the Domain
+   */
+  readonly domainName: string;
+  /**
+   * The key used to encrypt the Domain
+   */
+  readonly encryptionKey?: IKey;
+}
+
+export class Domain extends DomainBase implements IDomain, ITaggableV2 {
+  public static fromDomainAttributes(scope: Construct, id: string, attrs: DomainAttributes): IDomain {
+    class Import extends DomainBase {
+      public readonly domainArn = attrs.domainArn;
+      public readonly domainName = attrs.domainName;
+      public readonly encryptionKeyArn = attrs.encryptionKeyArn;
+      public readonly domainOwner = attrs.domainOwner;
+    }
+
+    return new Import(scope, id);
+  }
+
+  public static fromDomainArn(scope: Construct, id: string, domainArn: string): IDomain {
+    const domainResourceArnParts = Stack.of(scope).splitArn(domainArn, ArnFormat.SLASH_RESOURCE_NAME);
+    if (
+      domainResourceArnParts.resource !== 'domain' ||
+      domainResourceArnParts.account === undefined ||
+      domainResourceArnParts.resourceName === undefined
+    ) {
+      throw new Error(`Expected a domain ARN, but got ${domainArn}`);
+    }
+    return Domain.fromDomainAttributes(scope, id, {
+      domainArn,
+      domainName: domainResourceArnParts.resourceName,
+      domainOwner: domainResourceArnParts.account,
+    });
+  }
+
+  protected cfnResource: CfnDomain;
+  private cfnResourceProps: CfnDomainProps;
+
+  readonly cdkTagManager: TagManager;
+  readonly domainArn: string;
+  readonly domainName: string;
+  readonly encryptionKeyArn?: string;
+  readonly domainOwner: string;
+  /**
+   * Optional policy document that represents the resource policy of this repository
+   *
+   * If specified, addToResourcePolicy can be used to edit this policy.
+   * Otherwise this method will no-op.
+   */
+  protected readonly policy?: PolicyDocument;
+
+  constructor(scope: Construct, id: string, props: DomainProps) {
+    super(scope, id);
+    this.policy = new PolicyDocument();
+    this.cdkTagManager = new TagManager(TagType.KEY_VALUE, 'AWS::CodeArtifact::Domain');
+
+    const encryptionKey =
+      props.encryptionKey ??
+      new Key(this, 'key', {
+        description: `Key for CodeArtifact Domain ${props.domainName}`,
+        alias: `codeartifact-domain/${props.domainName}`,
+      });
+
+    this.cfnResourceProps = {
+      domainName: props.domainName,
+      encryptionKey: encryptionKey?.keyArn,
+      tags: this.cdkTagManager.renderTags(),
```

Comment:
Did I miss something? What is this doing?

Reply:
I thought I'd need to render the tags via the Tagmanager, but I guess this is done [implicitly](https://github.com/aws/aws-cdk/blob/b49032b3a6e549783b45492ffc76880fbcd58e68/packages/aws-cdk-lib/core/lib/cfn-resource.ts#L493-L504). :shrug: 

So I guess I'll remove it here...